x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java

```diff
@@ -51,12 +51,23 @@ public DataFrameIndexer(Executor executor,
 
     protected abstract Map<String, String> getFieldMappings();
 
+    /**
+     * Request a checkpoint
+     */
+    protected abstract void createCheckpoint(ActionListener<Void> listener);
+
     @Override
     protected void onStart(long now, ActionListener<Void> listener) {
         try {
             QueryBuilder queryBuilder = getConfig().getSource().getQueryConfig().getQuery();
             pivot = new Pivot(getConfig().getSource().getIndex(), queryBuilder, getConfig().getPivotConfig());
-            listener.onResponse(null);
+
+            // if run for the 1st time, create checkpoint
+            if (getPosition() == null) {
```

Comment:
could this be abstracted out and put into a method called isFirstRun()?

I am thinking that other actions will have to be take on the first run, and in the future (I assume), this will include something other than verifying the position is null.

If this is not done in this PR, it will probably still be done down the line in some other work :).